src/services/vaultTimeout.service.ts

```diff
@@ -71,12 +75,12 @@ export class VaultTimeoutService implements VaultTimeoutServiceAbstraction {
             return;
         }
 
-        // TODO update with vault timeout name and pivot based on action saved
-        const lockOptionSeconds = lockOption * 60;
+        const vaultTimeoutSeconds = vaultTimeout * 60;
         const diffSeconds = ((new Date()).getTime() - lastActive) / 1000;
-        if (diffSeconds >= lockOptionSeconds) {
-            // need to lock now
-            await this.lock(true);
+        if (diffSeconds >= vaultTimeoutSeconds) {
+            // Pivot based on the saved vault timeout action
+            await this.storageService.get<string>(ConstantsService.vaultTimeoutActionKey) === 'lock' ?
+                await this.lock(true) : await this.logOut();
```

Comment:
```suggestion
            const timeoutAction = await this.storageService.get<string>(ConstantsService.vaultTimeoutActionKey);
            timeoutAction === 'lock' ? await this.lock(true) : await this.logOut();
```

Reply:
Updated.  

What's the reason for this request (just curious)?

Reply:
It was just a suggestion that kept the conditional check line a bit cleaner IMO

Reply:
Fair and accurate. Thanks!